### ISSUE TITLE
utf8 library overrides string methods, breaks networking code

### ISSUE THREAD

**Issue by sharpobject** [KennyShields/LoveFrames#166]
Hi,

I notice that UTF-8 support has landed in LoveFrames. Great! Unfortunately, the utf8 library replaces a bunch of built-in functions with broken versions of those functions which crash when given reasonable inputs. You can see the offending lines here:

https://github.com/KennyShields/LoveFrames/blob/master/libraries/utf8.lua#L1032

I think the right way to fix this would be to not add anything to the global string object and to call the utf8 functions explicitly everywhere we mean to. If you'd prefer some other approach let me know.

**Comment by KennyShields** [comment]
Yeah, this is primarily why its disabled by default now. I brought this issue up with the person who added the library originally but he seemed to think there would be issues if the library were to be changed: https://github.com/KennyShields/LoveFrames/pull/153. As for a solution to this matter, I'm open to pretty much anything as long its stable and doesn't modify global functions like the current implementation.

**Comment by Stepets** [comment]
Hello!

@sharpobject can you show minimal crashing code example, please?

@KennyShields problem is to rewrite all `randomString:<string-function>()` code already presented in loveframes project. I'm to lazy for this.

**Comment by unek** [comment]
fixed it:

https://github.com/unek/LoveFrames/commit/dc50f9e750a2738fbe19540f14234fcd05e614cd?w=1
https://github.com/unek/LoveFrames/commit/c73cfe70ad6cb12faf473772d38ac4518771b500

**Comment by sharpobject** [comment]
```
> return ("\200"):len()
1
> require("utf8")
> return ("\200"):len()
./utf8.lua:109: UTF-8 string terminated early
stack traceback:
	[C]: in function 'error'
	./utf8.lua:109: in function 'utf8charbytes'
	./utf8.lua:194: in function <./utf8.lua:181>
	(...tail calls...)
	[C]: in ?
```

There are a bunch of other ways to get strings like this. For example, you could read some binary data from a socket, or if your server is written in lua and also uses utf8.lua, you could read some "unicode-encoded binary data" from a socket but happen to not read a whole unicode character.

**Comment by Stepets** [comment]
@sharpobject , I see, thanks for reply!
It was a really bad idea to reload all functions, my apologies.

For now I have already rewritten textinput. But both default skins doesn't support utf8, so I plan to make new one.

**Comment by unek** [comment]
@Stepets no changes are needed for skins to work properly with utf-8.

**Comment by Stepets** [comment]
@unek don't think so.
At least, what I noticed:
-masked textinputs has wrong number of displaying symbols

**Comment by unek** [comment]
@Stepets oh yes, that's right. you can make the skins use utf8.gsub instead of string.gsub.

**Comment by KennyShields** [comment]
Just a heads up: I'm creating a new ui library for LOVE, so I don't plan on adding proper utf8 support to Love Frames. The new library, however, will support utf8.

**Comment by unek** [comment]
@KennyShields that's cool. what's the progress so far?

**Comment by Stepets** [comment]
Hello @KennyShields !
Will it be opensourced?
Make it plz rendering ui elements to sprite or smth, it can be usefull for caching or displaying in-game.

**Comment by KennyShields** [comment]
@unek I have a lot of the core features working (things like instancing, new object template system, etc) as well as some basic objects. Currently working on implementing scrollable content.

@Stepets Yes, the source code will be libre. Also, I'm not entirely sure how rendering will be done yet, but thats always a possibility.

**Comment by unek** [comment]
@KennyShields please don't make the scrolling like it's currently done in loveframes.

**Comment by KennyShields** [comment]
@unek Well, my plan is to make scrollbars attachable to any object to allow their content to be scrolled. Can you be more specific about what you think is wrong with the way Love Frames handles it?

**Comment by unek** [comment]
@KennyShields currently :Scroll belongs to the scrollbar object and takes only one argument which is the amount of pixels the scrollbar needs to be moved.

I think :Scroll should take two arguments: amounts of pixels the content needs to be scrolled in each direction (the content, not the scrollbar). also :Scroll should belong to any scrollable object.
